### PR TITLE
removing impact warning messages 

### PR DESCRIPTION
**Pull request by jgaleotti**
Adding logic to avoid "unexpected handling on Initialization Action after repair" when it is not necessary

### CODE REVIEW

File: core/src/main/kotlin/org/evomaster/core/search/EvaluatedIndividual.kt [hunk 1]
```diff
@@ -783,6 +784,11 @@ class EvaluatedIndividual<T>(
         }
         mutatedGenes.addedInitializationGenes.addAll(diff.flatMap { it.seeTopGenes() })
 
+        if (addedInsertions!!.flatten().isEmpty()) {
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/search/EvaluatedIndividual.kt:787]: @man-zhang what would be the case of having `! diff.isEmpty()` but then this `if` is true? ie, the `diff` is not empty, but `addedInsertions`  is? 
would that be a bug in EM?

Comment (man-zhang) [reply core/src/main/kotlin/org/evomaster/core/search/EvaluatedIndividual.kt:787]: the difference might be due to db action repair. 
here, we aim to keep consistency between genes and their impacts.

Comment (man-zhang) [reply core/src/main/kotlin/org/evomaster/core/search/EvaluatedIndividual.kt:787]: Hi, @arcuri82 
I cannot comment on the code which has been merged.
Please check the line 777,

``val diff = individual.seeInitializingActions()
            .filter { !old.contains(it) && ((it is SqlAction && !it.representExistingData) || it is MongoDbAction) }``

I am not sure whether the `MongoDbAction` should be always considered as `diff`. 
Could you please have a check?

**Comment by man-zhang** [comment]
hi, @arcuri82  might need some discussion about impacts of MongoDb and DNS actions

**Comment by jgaleotti** [comment]
sound good to me!

**Comment by man-zhang** [comment]
sounds good to me as well. it works for me before 3pm Oslo time.

**Comment by arcuri82** [comment]
thx @man-zhang . @jgaleotti has created a meeting invitation

**Comment by arcuri82** [comment]
@man-zhang  can you take over this branch? and make sure MongoDB works fine with impact collection?

**Comment by jgaleotti** [comment]
Any updates on this PR?

**Comment by arcuri82** [comment]
@jgaleotti hi. waiting for some more fixes in other branches before resuming work on this one.

**Review by arcuri82 — APPROVED** [review]

**Comment by jgaleotti** [comment]
thxs!

JP

**Pull request merged**